corehq/apps/cloudcare/views.py

```diff
@@ -62,116 +65,129 @@ def insufficient_privilege(request, domain, *args, **kwargs):
-        else:
-            return "en"
+    @use_knockout_js
+    @use_bootstrap3
+    @use_datatables
+    @method_decorator(require_cloudcare_access)
+    @method_decorator(requires_privilege_for_commcare_user(privileges.CLOUDCARE))
+    def dispatch(self, request, *args, **kwargs):
+        return super(CloudcareMain, self).dispatch(request, *args, **kwargs)
+
+    def get(self, request, domain, urlPath):
+        try:
+            preview = string_to_boolean(request.REQUEST.get("preview", "false"))
+        except ValueError:
+            # this is typically only set at all if it's intended to be true so this
+            # is a reasonable default for "something went wrong"
+            preview = True
+
+        app_access = ApplicationAccess.get_by_domain(domain)
+
+        if not preview:
+            apps = get_cloudcare_apps(domain)
+            if request.project.use_cloudcare_releases:
+                # replace the apps with the last starred build of each app, removing the ones that aren't starred
+                apps = filter(
+                    lambda app: app.is_released,
+                    [get_app(domain, app['_id'], latest=True) for app in apps]
+                )
+                # convert to json
+                apps = [get_app_json(app) for app in apps]
+            else:
+                # legacy functionality - use the latest build regardless of stars
+                apps = [get_app_json(ApplicationBase.get_latest_build(domain, app['_id'])) for app in apps]
 
-    # default language to user's preference, followed by 
-    # first app's default, followed by english
-    language = request.couch_user.language or _default_lang()
-    
-    def _url_context():
-        # given a url path, returns potentially the app, parent, and case, if
-        # they're selected. the front end optimizes with these to avoid excess
-        # server calls
-
-        # there's an annoying dependency between this logic and backbone's
-        # url routing that seems hard to solve well. this needs to be synced
-        # with apps.js if anything changes
-
-        # for apps anything with "view/app/" works
-
-        # for cases it will be:
-        # "view/:app/:module/:form/case/:case/"
-
-        # if there are parent cases, it will be:
-        # "view/:app/:module/:form/parent/:parent/case/:case/
-        
-        # could use regex here but this is actually simpler with the potential
-        # absence of a trailing slash
-        split = urlPath.split('/')
-        app_id = split[1] if len(split) >= 2 else None
-
-        if len(split) >= 5 and split[4] == "parent":
-            parent_id = split[5]
-            case_id = split[7] if len(split) >= 7 else None
         else:
-            parent_id = None
-            case_id = split[5] if len(split) >= 6 else None
+            apps = ApplicationBase.view('app_manager/applications_brief', startkey=[domain], endkey=[domain, {}])
```

Comment:
yeah, but for this PR i might deem out of scope, this is all just whitespace since i had to make it a class based view to use the decorator